### PR TITLE
7️⃣ Lecture 7: Sigma Protocols

### PR DESCRIPTION
**Pull request by ZamDimon**
# Objective

This pull request adds all the materials related to Lecture 7: Sigma Protocols. We add examples and some generalizations of discrete-log-based Sigma protocols.

### CODE REVIEW

File: lectures/7-sigma.tex [hunk 1]
```diff
@@ -2,7 +2,483 @@
 
 \begin{document}
 
+\subsection{Schnorr's Identification Protocol}
+
+To better illustrate the Fiat-Shamir Transformation in practice, let us consider one of the most widely used signatures: Schnorr Signature. It is a simple and elegant protocol that allows one party to prove to another party that it knows a discrete logarithm of a given element.
```
Comment (Alex-distributed-lab) [inline lectures/7-sigma.tex]: Formally, the signature differs from the identification protocol. The identification protocol is designed only for secret knowledge and random challenges. The signature must cover some data proving its integrity (additionally to secret knowledge).

File: lectures/7-sigma.tex [hunk 2]
```diff
@@ -2,7 +2,483 @@
...
 
+\subsection{Schnorr's Identification Protocol}
+
+To better illustrate the Fiat-Shamir Transformation in practice, let us consider one of the most widely used signatures: Schnorr Signature. It is a simple and elegant protocol that allows one party to prove to another party that it knows a discrete logarithm of a given element. 
+
+Let us formalize it using theory from \Cref{section:relations_and_languages}. Suppose $\mathbb{G}$ is a cyclic group of order $q$ with a generator $g$. Then, the relation and language being considered are:
+\begin{equation*}
+    \mathcal{R} = \{(u, \alpha) \in \mathbb{G} \times \mathbb{Z}_q: u = g^{\alpha}\}, \; \mathcal{L}_{\mathcal{R}} = \{u \in \mathbb{G}: \exists \alpha \in \mathbb{Z}_q: u = g^{\alpha}\}
+\end{equation*}
+
```
Comment (Alex-distributed-lab) [inline lectures/7-sigma.tex:13]: If it makes sense, in the future can be rewritten to notation using scalars and EC Points

**Review by Alex-distributed-lab — APPROVED** [review]
Looks good

File: lectures/7-sigma.tex (hunk 1 above)
Comment (ZamDimon) [reply lectures/7-sigma.tex]: Yeah, very good point, will correct it later (probably after the lecture)

File: lectures/7-sigma.tex (hunk 2 above)
Comment (ZamDimon) [reply lectures/7-sigma.tex:13]: I believe we can return to that after finishing the lecture notes. Typical scientific literature uses multiplicative notation in 80+% of cases (assuming that the underlying group is an elliptic curve nonetheless). However, additive notation may be easier to understand and deal with. 

Worth keeping this in mind, thanks!

**Pull request merged**